src/coreclr/jit/morph.cpp

```diff
@@ -13969,6 +13977,143 @@ GenTree* Compiler::fgOptimizeBitwiseAnd(GenTreeOp* andOp)
     return nullptr;
 }
 
+//------------------------------------------------------------------------
```

Comment:
I think it would be valuable to add a `TODO-Casts: consider unifying this function with "optNarrowTree"` here somewhere.

This function is essentially doing a subset of what `optNarrowTree` does "manually".